blockprod/src/detail/mod.rs

```diff
@@ -328,13 +321,33 @@ impl BlockProduction {
         let (job_remover_func, end_confirm_receiver) = self.job_manager.make_job_stopper_function();
         let _job_remover = OnceDestructor::new(move || job_remover_func(job_key));
 
+        let mut previous_attempt = timestamp_at_start;
+        let mut previous_consensus_status =
```

Comment:
Why do we need to keep updating this if we're targeting one, specific previous block?

Reply:
As above, this has now changed.

Reply:
I've removed the checking of consensus here since we should be doing a MAX block timestamp check regardless of the consensus we're currently configured for.

Reply:
An issue I see here now from this change is that PoW block timestamps will be `prev+1` as we don't advance. Not sure what you want as a design decision for this, but an option would be to add back in the `block_timestamp++` for the loop, or to have PoW consensus use `current_timestamp i.e NOW` internally.

Reply:
This is gone now.